fms/models/__init__.py

```diff
@@ -70,41 +70,74 @@ def list_variants(architecture: str):
 
 def __maybe_infer_model_variant(
     architecture: str,
-    variant: str,
+    variant: Optional[str],
     model_path: Optional[str],
     source: Optional[str],
     **kwargs,
 ) -> Tuple[str, str, Optional[str], Optional[str], Dict[str, Any]]:
     """Infer the model variant configuration from different sources, currently only supported sources are hf"""
     extra_kwargs = kwargs
 
-    if architecture in ("hf_pretrained", "hf_configured"):
+    if architecture in ("hf_pretrained", "hf_configured", "hf_inferred"):
         from fms.models.hf.utils import _infer_model_configuration  # type: ignore
 
-        logger.info(f"inferring model configuration from {variant}")
         is_hf_pretrained = architecture == "hf_pretrained"
-        if is_hf_pretrained:
-            if model_path is not None or source is not None:
+        is_hf_configured = architecture == "hf_configured"
+        is_hf_inferred = architecture == "hf_inferred"
+
+        required_args: List[Any] = []
+        banned_args: List[Any] = []
+        if is_hf_pretrained or is_hf_configured:
+            logger.info(f"inferring model configuration from {variant}")
+            if is_hf_pretrained:
+                required_args = [variant]
+                banned_args = [model_path, source]
+            if is_hf_configured:
+                required_args = [variant, model_path, source]
+                banned_args = []
+        elif architecture == "hf_inferred":
```

Comment:
this could probably be with the other 2 conditionals above

Reply:
simplified the logic